### DISCUSSION TITLE
Installing SpaCy with cuda on Jetson Nano.

### DISCUSSION THREAD

**Discussion by apollowebdesigns** [explosion/spaCy#9367] [Help: Installation] [answered]
Hello, I have really enjoyed working with spaCy and found it useful to work with for NLP problems.

I was trying to install the latest version of spaCy on my Jetson Nano but was unable to install via trying various routes, including - 

https://gist.github.com/lucidbeaming/293f26494f9514e9546e0d7856ae0b20

I was wondering if there was an easier/straightforward way that I have overlooked?

Thanks for any comments in advance.

**Comment by polm** [comment]
I'd never heard of the Jetson Nano, it looks like it's like some kind of GPU raspi-like from Nvidia? The main thing is it's arm64 and we don't distribute wheels for that, so you'll have to build from source, and I'm not sure that'll work. If you look at [this issue](https://github.com/explosion/cython-blis/issues/9#issuecomment-929122331) it looks like maybe people have gotten it working in the past.

We have been working a little on arm64 support but basically just for Macs as far as I'm aware.

**Reply by apollowebdesigns** [reply]
Yes, Jetson Nano is like a Pi but with GPU.

More of the issue when building locally is a dependency called thinc which has issues installing as it requires versions 8.0.10 and the greatest I can get is 8.0.8

I will try that project next, unless you have any suggestions to install the thinc dependency?

**Comment by adrianeboyd** [comment, marked as answer]
Support for `spacy` and its immediate dependencies for arm64 / aarch64 on mac os and linux running on CPU should be fine. You should be able to run `pip install spacy` to compile and install everything in one step. (It's going to be slow, though. Starting from scratch it takes around an hour on a raspberry pi 4.)

For just `spacy` on CPU I would recommend trying to use `conda` instead of `pip`, since there should be binary `conda-forge` packages for `linux-aarch64` for all the dependencies. Start with the miniforge aarch64 installer (https://github.com/conda-forge/miniforge#miniforge3) and then follow the directions for spacy with conda: https://spacy.io/usage

If you're very lucky, then `conda install -c conda-forge cupy` will just work and you can run non-torch models on GPU, too, but I have no idea if it will actually work in the end. We've only tested on a raspberry pi and apple m1, so we haven't tested anything related to CUDA.

If you want to use `spacy-transformers` and `torch`:

* Follow the link in that script to get more recent pytorch releases from nvidia
* You can probably also just `pip install sentencepiece` with the most recent version

I'm not sure the conda venv will work in combination with the torch wheels from nvidia, though, and I'm not sure that you'll have much success using larger transformer models with such limited resources.

In any case, the small models that are designed to perform well on CPU like `en_core_web_sm` should be fine. Slightly larger models like `en_core_web_md` should be similar in speed, but just require more RAM. `en_core_web_lg` with its large vector table might require too much RAM to be usable.

If you can post specific error messages, we can try to help, but without more details there's not much else we can suggest.
